VMEncryption/main/ExtensionParameter.py

```diff
@@ -174,6 +174,13 @@ def _is_kv_equivalent(self, a, b):
             if b[-1] == '/': b = b[:-1]
         return a==b
 
+    def cmk_changed(self):
+        if (self.KeyEncryptionKeyURL or self.get_kek_url()) and \
+           (not self._is_kv_equivalent(self.KeyEncryptionKeyURL, self.get_kek_url())):
+           self.logger.log('Current config KeyEncryptionKeyURL {0} differs from effective config KeyEncryptionKeyURL {1}'.format(self.KeyEncryptionKeyURL, self.get_kek_url()))
+           return True
```

Comment:
There could be scenarios where customers want to retry the rotation in case of previous failure. For that case they use sequence number for running the operation again. To support this , shouldn't the check to update the encryption settings to luks header also have the logic to check whether the sequence number changed or not?

Reply:
encryption setting is stored locally only if rotation is successful. 
in case of failure cmk_changed will always returns true.